### PR TITLE
Extract ArticleSection for readability

### PR DESCRIPTION
**Pull request by stephanebisson**

### CODE REVIEW

File: src/components/Article.js [hunk 1]
```diff
@@ -13,33 +13,27 @@ const ArticleBody = memo(({ content }) => {
...
   useEffect(() => {
-    setNavigation(0)
-  }, [article])
+    // todo: the enter key should only be available on the
+    // first page of the first section
+    const text = hasActions ? i18n.i18n('centerkey-select') : ''
+    const handler = hasActions ? onKeyCenter : null
+    softkey.dispatch({ type: 'setCenterText', value: text })
+    softkey.dispatch({ type: 'setOnKeyCenter', event: handler })
+  }, [hasActions])
```
[the remark this answers is not in the document]
Comment (stephanebisson) [reply src/components/Article.js]: This function depends on `hasActions` on lines 31 and 32. If I remove it from the dependencies on line 35, the function does not re-run when we change section and the center key text does not update.

It is important for the softkeys to be context-sensitive. It should not display "ENTER" in the context where pressing enter will do nothing.

File: src/components/Article.js [hunk 2]
```diff
@@ -13,33 +13,27 @@ const ArticleBody = memo(({ content }) => {
...
-    softkey.dispatch({ type: 'setOnKeyLeft', event: onKeyLeft })
-    softkey.dispatch({ type: 'setOnKeyCenter', event: onKeyCenter })
+    setNavigation(0)
   }, [])
 
   useEffect(() => {
-    setNavigation(0)
-  }, [article])
+    // todo: the enter key should only be available on the
+    // first page of the first section
```
[the remark this answers is not in the document]
Comment (stephanebisson) [reply src/components/Article.js]: Right, we can keep the handler and decide what it does, if anything, based on getCurrent()

[the remark this answers is not in the document]
Comment (hueitan) [reply src/components/Article.js]: then we can remove this todo comment

[the remark this answers is not in the document]
Comment (stephanebisson) [reply src/components/Article.js]: Not quite, the soft keys are contextual to what is shown on the screen. When you go to the second page, the actions are not visible so the text should be blank and the enter key should do nothing.

To do that, I think we would need to bring the Pager or useScroll inside of the ArticleSection.

[the remark this answers is not in the document]
Comment (hueitan) [reply src/components/Article.js]: I got what you mean now, there is a case there is no action and no text is selected, so the center text will be blank.

**Pull request merged**